### ISSUE TITLE
Implement Predictive Coding local Hebbian learning rules

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1555]
## Summary

Implement the "slow" learning phase of Predictive Coding — local Hebbian-like weight update rules that adjust synaptic weights based on prediction errors after inference has settled. This is the key differentiator from standard backpropagation: updates depend only on locally available information (pre-synaptic activity and post-synaptic error).

Part of #1549
Depends on: #1554 (PC inference engine)

### 1. Create `src/predictiveCoding/PredictiveCodingLearning.ts`

Implement the PC weight update rule:

```
After inference has settled for a data sample:
  For each synapse with weight W connecting neuron j → neuron i:
    ΔW = η_learn · f'(a(i)) · ε(i) · x(j)
    
  Where:
    η_learn = PC learning rate (from config)
    f'(a(i)) = derivative of squash function at pre-activation of target neuron
    ε(i) = prediction error at target neuron (from inference)
    x(j) = activation of source neuron (latent value after settling)
```

This is a local Hebbian rule: each synapse update depends only on:
- The error at its target neuron
- The activation of its source neuron
- The local derivative of the activation function

### 2. Bias Updates

Similarly, bias updates use local prediction errors:
```
Δb(i) = η_learn · f'(a(i)) · ε(i)
```

### 3. Batch Accumulation

- Accumulate weight/bias gradients across a mini-batch of samples
- Apply averaged updates after the batch completes
- This maintains the two-timescale separation: many inference steps per weight update

### 4. Integration with Existing Weight Constraints

- Respect existing weight precision constraints (minimum 1 plank unit = 1e-7)
- Respect synapse type constraints (positive, negative, condition)
- Apply `maximumWeightAdjustmentScale` from existing training config if configured
- Ensure weight updates don't violate topology constraints

## Acceptance Criteria

- [ ] `PredictiveCodingLearning.ts` implements local Hebbian weight updates
- [ ] Bias updates implemented alongside weight updates
- [ ] Batch accumulation mode works correctly
- [ ] Weight constraints (precision, type, scale) respected
- [ ] Unit tests including:
  - Known gradient: hand-computed weight update matches implementation
  - Learning reduces energy: verify total prediction error decreases over training epochs
  - Weight constraints: verify updates respect precision and type bounds
  - Batch vs single-sample: verify batch averaging produces correct results
  - Squash derivative: verify correct derivatives for multiple activation functions (RELU, TANH, SIGMOID)
- [ ] All existing tests pass unchanged
- [ ] `./quality.sh` passes
- [ ] Australian English spelling throughout

## Coding Practices Reminder
- TDD: Write a test with hand-computed expected weight deltas before implementing the update rule
- DRY: Reuse existing squash function derivative implementations from `src/methods/activations/`
- KISS: Single learning rate for all layers initially; per-layer rates can be a future enhancement

## References
- Whittington & Bogacz (2017): "An Approximation of the Error Backpropagation Algorithm in a Predictive Coding Network with Local Hebbian Synaptic Plasticity"
- Millidge et al. (2022): "Predictive Coding Approximates Backprop Along Arbitrary Computation Graphs"
